app/public/src/game/components/wanderers-manager.ts

```diff
@@ -198,9 +203,8 @@ export default class WanderersManager {
     sprite.sprite.setInteractive()
     sprite.sprite.on("pointerdown", (pointer: Phaser.Input.Pointer) => {
       if (clicked) return
-      clicked = true
-      tweens.forEach((tween) => tween.destroy())
-      onClick(wanderer, sprite, pointer, tween)
+      if ((clicked = onClick(wanderer, sprite, pointer, tweens)))
```

Comment:
avoid assignments inside if conditions, this is a bug-prone pattern

```suggestion
clicked = onClick(wanderer, sprite, pointer, tweens)
      if (clicked)
```